src/libstd/collections/hash/map.rs

```diff
@@ -1271,6 +1299,12 @@ pub struct Drain<'a, K: 'a, V: 'a> {
     inner: iter::Map<table::Drain<'a, K, V>, fn((SafeHash, K, V)) -> (K, V)>
 }
 
+/// Mutable HashMap values iterator.
+#[unstable(feature = "map_values_mut", reason = "recently added", issue = "32551")]
+pub struct ValuesMut<'a, K: 'a, V: 'a> {
+    inner: Map<IterMut<'a, K, V>, fn((&'a K, &'a mut V)) -> &'a mut V>
```

Comment:
Actually, now that I think about it some more, using `Map` makes `K` incorrectly invariant. This probably doesn't matter in practice, but I'd still be more comfortable implementing that mapping by hand and getting the correct covariance.